`mvn clean` does not work in js directory

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

When performing rebuild (e.g. due to rebase), `mvn clean` would not clean `js` subdirectories which sometimes leads to errors in subsequent `mvn install` command.

### Version

main

### Expected behavior

`mvn clean install` rebuilds the project flawlessly after rebase.

### Actual behavior

`mvn clean install` fails sometimes with error. E.g. the following one, but varies depending on changed code after rebase:

```
[INFO] --- frontend:1.14.2:pnpm (run-build) @ keycloak-admin-ui ---
[INFO] Running 'pnpm run build' in /home/hmlnarik/src/keycloak/js/apps/admin-ui
[INFO] 
[INFO] > admin-ui@ build /home/hmlnarik/src/keycloak/js/apps/admin-ui
[INFO] > wireit
[INFO] 
[INFO] ⠋ Analyzing
[INFO] ⠙   0% [0 / 4] [0 running]⠹⠸⠼⠴   0% [0 / 4] [1 running] ../../libs/keycloak-admin-client:build⠦  25% [1 / 4] [1 running] ../../libs/keycloak-admin-client:build⠧                                                                 ❌ [../../libs/keycloak-admin-client:build] exited with exit code 1. Output:
[INFO] 
[INFO] node:internal/modules/cjs/loader:1080
[INFO]   throw err;
[INFO]   ^
[INFO] 
[INFO] Error: Cannot find module '/home/hmlnarik/src/keycloak/js/node_modules/.pnpm/typescript@5.3.2/node_modules/typescript/bin/tsc'
[INFO]     at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
[INFO]     at Module._load (node:internal/modules/cjs/loader:922:27)
[INFO]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:86:12)
[INFO]     at node:internal/main/run_main_module:23:47 {
[INFO]   code: 'MODULE_NOT_FOUND',
[INFO]   requireStack: []
[INFO] }
[INFO] 
[INFO] Node.js v18.18.2
[INFO] ⠇
[INFO] ❌ 1 script failed. ELIFECYCLE  Command failed with exit code 1.
```

### How to Reproduce?

1. checkout some older version, e.g. 22
2. `mvn clean install`
3. rebase to latest main
4. `mvn clean install`

### Anything else?

_No response_

--- a/js/pom.xml
+++ b/js/pom.xml
@@ -51,6 +51,29 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-clean-plugin</artifactId>
+                <configuration>
+                    <followSymLinks>false</followSymLinks>
+                    <filesets>
+                        <fileset>
+                            <directory>${basedir}</directory>
+                            <includes>
+                                <include>**/.wireit/**</include>
+                                <include>**/node_modules/**</include>
+                            </includes>
+                        </fileset>
+                        <!-- include all non-maven projects here as well -->
+                        <fileset>
+                            <directory>apps/keycloak-server</directory>
+                            <includes>
+                                <include>server/**</include>
+                            </includes>
+                        </fileset>
+                    </filesets>
+                </configuration>
+            </plugin>
         </plugins>
     </build>
 </project>

